history-service/src/redis_stream_adapter/consumer.d.ts

```diff
@@ -0,0 +1,3 @@
+declare interface IStreamConsumer {
+  runConsumer(call: (response: { [property: string]: string }) => void);
```

Comment:
It would perhaps be better if you separate it into
- `run()` - Start the consumer waiting thread
- `addListener(call: (response) => {})` - Hook a listener to this consumer